### DISCUSSION TITLE
[Typescript] Accessing theme nested colors objects

### DISCUSSION THREAD

**Discussion by giubatt** [system-ui/theme-ui#1652] [Q&A] [answered]
I'm having trouble with accessing a nested color from the theme object in my component, using `useThemeUI` or a function inside `sx`

If I define my theme like this for example
```
export const theme: Theme = {
  colors: {
    primary: {
      __default: "blue",
      light: "lightblue",
    },
  },
}
```
![image](https://user-images.githubusercontent.com/12552164/114578683-7f7c4000-9c53-11eb-894e-3c38e892d4d4.png)

There are no type errors, the new theme typings work correctly.

However, when I try to use that theme object, I can't figure out a way to get the `primary.light` without typescript yelling at me:
![image](https://user-images.githubusercontent.com/12552164/114579692-7cce1a80-9c54-11eb-8dc8-594517c5a033.png)
I tried the new `rawColors` as well as just `colors`, both have the same type definitions

Is there a better way to do this? Usually I just use "primary.light" in my components, but sometimes I need to get the raw color value to use somewhere

**Comment by franklinjavier** [comment, marked as answer]
https://theme-ui.com/guides/typescript
